### ISSUE TITLE
add way to specify custom shaders

### ISSUE THREAD

**Issue by bdpedigo** [CAVEconnectome/nglui#32]
would be nice to have a way to pass in custom shaders, e.g. for a skeleton in the segmentation layer. here's an example workaround I did just by modifying the JSON

```
shader = """
void main() {
    float compartment = vCustom2;
    vec4 uColor = segmentColor();
    emitRGB(uColor.rgb*0.5 + vec3(0.5, 0.5, 0.5) * compartment);
}
"""
skel_rendering_kws = {
    "shader": shader,
    "mode2d": "lines_and_points",
    "mode3d": "lines",
    "lineWidth3d": 1,
}

state_dict["layers"][1]["skeletonRendering"] = skel_rendering_kws

statebuilder.StateBuilder(base_state=state_dict, client=client).render_state(
    return_as="html"
)
```

**Comment by ceesem** [comment]
Just adding the shader as text is trivial, but I would love some assistance/feedback to add helper functions to build a collection of useful shader primitives like this one.

**Comment by bdpedigo** [comment]
would stuff like "color by component" fall under your view of primitives? any other specific examples you can think of?

**Comment by ceesem** [comment]
Yeah, "color by $property" is definitely one — it would be nice to have something that reads the info file and can predict/validate what values might be accessible. I'm also thinking more broadly than skeletons, since annotation layers and such also can have useful shaders.

**Comment by ceesem** [comment]
In this case, for example, how on earth would someone know that it's vCustom2 and not vCustom1 or... whatever?

**Comment by ceesem** [comment]
Skeleton shader support is now integrated into SegmentationLayerConfig objects, although smart defaults are not yet in there.
